### PR TITLE
LTTP: Move LTTP spoiler writing out of core

### PR DESCRIPTION
**Pull request by alwaysintreble**
## What is this fixing or adding?
Removed some special casing in the spoiler writing that was unnecessary/didn't actually do anything, and moved all the LTTP info we actually want output to its world.

## How was this tested?
Generated a few games and looked at the spoiler

### CODE REVIEW

File: test/general/TestItems.py [hunk 1]
```diff
@@ -44,8 +44,6 @@ def testItemNameGroupConflict(self):
     def testItemCountGreaterEqualLocations(self):
         for game_name, world_type in AutoWorldRegister.world_types.items():
 
-            if game_name in {"Final Fantasy"}:
-                continue
```
Comment (black-sliver) [inline test/general/TestItems.py:48]: always leaking stuff :D

**Review by black-sliver — COMMENTED** [review]
Given the `+127 −191` figure, did you run it through aproller with all teh yamls to check if you removed something that was required?

**Comment by black-sliver** [comment]
Also, you sure removing stuff from `to_json` and `to_file` is fine? I don't know what they are used for, so @Berserker66 ?

File: test/general/TestItems.py (hunk 1 above)
Comment (alwaysintreble) [reply test/general/TestItems.py:48]: odd i dropped that locally. might need to force push.

**Comment by alwaysintreble** [comment]
I don't know that it's fine, but I did generate seeds and it worked. I didn't test WebHost though so it might be necessary there somewhere.

**Comment by alwaysintreble** [comment]
I fully removed `to_json` and did some testing and everything seems to work fine. I also couldn't find any references to it so I'm not sure what it was there for. Have it committed locally, and can push on request, but not fully sure that it's safe to do so.

**Comment by Berserker66** [comment]
to_json is what ER used to output its data to alttpr

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**